### ISSUE TITLE
Question about data preparation & train

### ISSUE THREAD

**Issue by lupengliu** [anicolson/DeepXi#12]
Anicolson, thanks for your great effort.

I used Tesla P40 for training. However, I found the time-cost in training is too long with mbatch=10.

I have some questions about data preparation & training step:
1. Do you use all of the training sets ( 74250 utterances) in the training step?
2. How long does 1 epoch spend to train?
3. Could you please share the device info for training?

Thanks a lot!

**Comment by anicolson** [comment]
Hi, 1 epoch consists of the 74250 clean speech recordings, where each clean speech recording is mixed with a random section of a randomly selected noise recording at a randomly selected SNR level. 1 epoch takes around 40 minutes using a GTX 1080 Ti or a TITAN X. These have 12-13GB of memory, and the mini-batch size is chosen to maximize the use of this. If you have more GPU memory available than this, a larger mini-batch size may reduce the amount of time required per epoch.

Hope this helps,
Aaron
________________________________

**Comment by lupengliu** [comment]
Hi, thank you for your reply!

I have used the same training dataset with you, and try to continuouse the training step (e.g. train from epoch 175). However, I got an even worse result in the succeded training steps.

Could you share your hyper-params(for example the learning rate and so on) of training 3d model on epoch 175? I'm trying to find the difference between my training script and yours. 


Thanks a lot!

**Comment by anicolson** [comment]
The hyperparameters used are in args.py. The default parameters for the adam optimizer are used. If you are using an identical training set, then epoch 175 is generally where the validation error stops improving. You could try using a smaller learning rate, this will require some modifications to the code.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
